homeassistant/components/ecobee/fan.py

```diff
@@ -0,0 +1,161 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_platform
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, ECOBEE_MODEL_TO_NAME, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ATTR_VENTILATOR_MIN_ON_TIME_HOME = "ventilator_min_on_time_home"
+ATTR_VENTILATOR_MIN_ON_TIME_AWAY = "ventilator_min_on_time_away"
+ATTR_IS_VENTILATOR_TIMER_ON = "is_ventilator_timer_on"
+
+SERVICE_SET_VENTILATOR_MIN_ON_TIME_HOME = "set_ventilator_min_on_time_home"
+SERVICE_SET_VENTILATOR_MIN_ON_TIME_AWAY = "set_ventilator_min_on_time_away"
+SERVICE_SET_VENTILATOR_TIMER = "set_ventilator_timer"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ecobee thermostat ventilator entity."""
+    data = hass.data[DOMAIN]
+    entities = []
+    _LOGGER.debug("Adding ventilators (if present)")
+    for index in range(len(data.ecobee.thermostats)):
+        thermostat = data.ecobee.get_thermostat(index)
+        if thermostat["settings"]["ventilatorType"] != "none":
+            _LOGGER.debug("Adding 1 ventilator")
+            entities.append(EcobeeVentilator(data, index))
+
+    async_add_entities(entities, True)
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_SET_VENTILATOR_MIN_ON_TIME_HOME,
+        {
+            vol.Required(ATTR_VENTILATOR_MIN_ON_TIME_HOME): vol.Coerce(int),
+        },
+        "set_ventilator_min_on_time_home",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_VENTILATOR_MIN_ON_TIME_AWAY,
+        {
+            vol.Required(ATTR_VENTILATOR_MIN_ON_TIME_AWAY): vol.Coerce(int),
+        },
+        "set_ventilator_min_on_time_away",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_VENTILATOR_TIMER,
+        {
+            vol.Required(ATTR_IS_VENTILATOR_TIMER_ON): cv.boolean,
+        },
+        "set_ventilator_timer",
+    )
+
+
+class EcobeeVentilator(FanEntity):
+    """A ventilator class for an ecobee thermostat with ventilator attached."""
+
+    def __init__(self, data, thermostat_index):
+        """Initialize ecobee ventilator platform."""
+        self.data = data
+        self.thermostat_index = thermostat_index
+        self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
+        ecobee_name = self.thermostat["name"]
+        self._attr_name = f"{ecobee_name} Ventilator"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.thermostat["identifier"])},
+            manufacturer=MANUFACTURER,
+            model=ECOBEE_MODEL_TO_NAME.get(self.thermostat["modelNumber"]),
+            name=self.name,
```

Comment:
```suggestion
            name=self._attr_name,
```
You might want to implement [this](https://developers.home-assistant.io/docs/core/entity?_highlight=has_en#has_entity_name-true-mandatory-for-new-integrations) right away